### ISSUE TITLE
Having troubles with Could not start process

### ISSUE THREAD

**Issue by Dreando** [flapdoodle-oss/de.flapdoodle.embed.mongo#294]
Hello,

so I saw a few different issues like this, but none of the proposed solutions turned to be helpful for me.

I have a project with primary datasource postgres and many existing integration spring boot tests (that use testcontainers postgresql). Recently we wanted to add a mongodb as a secondary DS for other purposes and I am hitting the wall when it comes to testing.

I added the floopdoodle mongodb artifact and the tests have stopped passing. Always at least single of them does not pass. It seams to be random which one. The failing test always says:

`java.io.IOException: Could not start process: <EOF>
`
and it was never the one that indeed tests mongo components, it works,but just random one. 

A few information:
```
spring boot version: 2.1.3.RELEASE
flapdoodle version: 2.2.0
```

application-test.properties:
```
spring.mongodb.embedded.version=3.6.0
spring.mongodb.embedded.features=sync_delay,no_http_interface_arg,only_with_ssl
```

the version 3.6.0 is indeed a bit old, but I wanted it to be same that we get from amazon document db.

Situation is the same on my mac 10.14.5 and on our CI which is Ubuntu 16.04.6 LTS (Xenial Xerus)

We're on openjdk 11

What could be important is that we're using JUnit5 with following properties:
```
junit.jupiter.testinstance.lifecycle.default=per_class
junit.jupiter.execution.parallel.enabled=true
```

Does anyone have any idea how to resolve this situation?

Thanks,
MK

**Comment by Loki-Afro** [comment]
I guess `junit.jupiter.execution.parallel.enabled=true`could be the issue here - don't know about the current spring stuff - but does it start a new mongodb per test or per class?

There is maybe a race condition when finding a random network port to use - although that never happened for me.

anyway maybe you can create some maven-surefire-plugin magic that these tests get executed sequential if that is really the problem

**Comment by Dreando** [comment]
Quick update: after digging for two days I overlooked one thing:
`spring.data.mongodb.port=0`
It did the thing! It probably could not start the process due to default port being in use, which anyways make me surprised as despite the fact of parallel execution, the tests seem to go _almost_ sequentially...

Anyways what I could see as an improvement would be to me this illegal argument saying a little bit more what's happening under the hood. I saw already a few different tickets and stackoverflow questions regarding this, most of them for different reasons.

In this situation, the ticket can be closed.

Thanks,
MK
